packages/react-enty/src/RequestHook.js

```diff
@@ -41,16 +42,19 @@ export default function RequestHookFactory(context: *, config: RequestHookConfig
 
         responseRef.current = response;
 
-        let onRequest = useCallback((payload) => {
+        let onRequest = useCallback((payload, {returnResponse = false} = {}) => {
             const responseKey = generateResultKey(payload);
             setResponseKey(responseKey);
-            return dispatch(requestAction(payload, {responseKey}))
-                .then(() => {
-                    if(mounted.current) {
-                        return responseRef.current;
-                    }
-                })
-                .catch(() => {});
+            const pending = dispatch(requestAction(payload, {responseKey}));
+
+
+            if (isObservable(pending) && returnResponse) {
+                return pending;
+            }
+
+            return returnResponse
+                ? pending.then(() => mounted.current ? responseRef.current : undefined)
+                : pending.then(() => {}).catch(() => {});
```

Comment:
If I have an observable and returnResponse is false, won't the execution end up here and this line will treat the observable like it's got a .then() method and break?

Reply:
Also the .then(() => {}) can probably be removed safely

Reply:
Also does the fact that returnResponse defaults to false mean that the useAutoRequest hook is redundant now, and useEffect can just be used instead?

(Learning about that void trick might also mean that too)